configs/projects/_shared-compiler-settings.rb

```diff
@@ -0,0 +1,24 @@
+# Define default CFLAGS and LDFLAGS for most platforms, and then
+# tweak or adjust them as needed.
+proj.setting(:cppflags, "-I#{proj.includedir} -I/opt/pl-build-tools/include")
+proj.setting(:cflags, "#{proj.cppflags}")
+proj.setting(:ldflags, "-L#{proj.libdir} -L/opt/pl-build-tools/lib -Wl,-rpath=#{proj.libdir}")
+
+# Platform specific overrides or settings, which may override the defaults
+
+# Harden Linux ELF binaries by compiling with PIE (Position Independent Executables) support,
+# stack canary and full RELRO.
+# We only do this on platforms that use their default OS toolchain since pl-gcc versions
+# are too old to support these flags.
+
+if((platform.is_sles? && platform.os_version.to_i >= 15) ||
+      (platform.is_el? && platform.os_version.to_i == 8 && platform.architecture !~ /ppc64/) ||
```

Comment:
Fot future proofing this, shouldn't we want `platform.os_version.to_i >= 8`?

Reply:
As we discussed earlier these changes doesn't work on EL and we have a separate ticket to fix this: https://perforce.atlassian.net/browse/PA-6896